### PR TITLE
feat(avatar): add label prop for alternative text & aria-label

### PR DESCRIPTION
**Pull request by anveshmekala**
**Related Issue:** #5564 

## Summary

This PR will add a `label` property in `calcite-avatar` which is used to provide context for Assistive Technology users.

### CODE REVIEW

File: src/components/avatar/avatar.tsx [hunk 1]
```diff
@@ -39,6 +39,9 @@ export class Avatar {
   /** Specifies the unique id of the user. */
   @Prop({ reflect: true }) userId: string;
 
+  /** When `thumbnail` is defined, specifies alternate text for the image. Otherwise specifies accessible label for the component.*/
```
[the remark this answers is not in the document]
Comment (geospatialem) [reply src/components/avatar/avatar.tsx:42]: For [line 33](https://github.com/Esri/calcite-components/pull/6910/files#diff-271a691b8124e6a04af47d1161ce8cf249434aba3c2873c12298f1eed9c2a82fR33), can it reflect the support for non-images?

> Specifies the full name of the user. When `label` and `thumbnail` are not defined, specifies the accessible name for the component.

**Review by geospatialem — COMMENTED** [review]
Thanks for taking a look so fast! 🐎 A few recommendations below for support of `aria-label` and doc considerations.

**Review by geospatialem — APPROVED** [review]
🏆 Sounds great in JAWS 🦈 and NVDA 🟣

**Review by jcfranco — APPROVED** [review]
👀👀👀👀👀👀👀👀👀👀👀👀👀👀👀👀👀👀👀👀👀👀👀
👀🖥👀👀👀🖥👀🖥🖥🖥👀👀🖥🖥🖥👀🖥🖥🖥🖥👀🖥👀
👀🖥🖥👀👀🖥👀👀🖥👀👀🖥👀👀👀👀🖥👀👀👀👀🖥👀
👀🖥👀🖥👀🖥👀👀🖥👀👀🖥👀👀👀👀🖥🖥🖥👀👀🖥👀
👀🖥👀👀🖥🖥👀👀🖥👀👀🖥👀👀👀👀🖥👀👀👀👀👀👀
👀🖥👀👀👀🖥👀🖥🖥🖥👀👀🖥🖥🖥👀🖥🖥🖥🖥👀🖥👀
👀👀👀👀👀👀👀👀👀👀👀👀👀👀👀👀👀👀👀👀👀👀👀

**Pull request merged**